### PR TITLE
Add dropout test

### PR DESCRIPTION
**Pull request by jla524**
Create a tensor of shape (100,000,) and add a dropout with rate=0.01. We can then count the number of zeros in this tensor and expect it to be around 100,000 * 0.01 = 1000 with some tolerance.

### CODE REVIEW

**Comment by jla524** [comment]
I’m not sure why the unit tests failed ._. From the log, it seems like `non_zeros` are completely off.

File: test/test_tensor.py [hunk 1]
```diff
@@ -64,6 +64,15 @@ def test_pytorch():
     for x,y in zip(test_tinygrad(), test_pytorch()):
       np.testing.assert_allclose(x, y, atol=1e-5)
 
+  @unittest.skipUnless(not DEFAULT_DEVICE, "float64 not supported on GPU")
```
Comment (geohot) [inline test/test_tensor.py:67]: You need float64 for this?

File: tinygrad/tensor.py [hunk 2]
```diff
@@ -295,7 +294,7 @@ def _pool2d(self, py, px):
 
   def avg_pool2d(self, kernel_size=(2,2)):
     return self._pool2d(*kernel_size).mean(axis=(3,5))
-
+
```
Comment (geohot) [inline tinygrad/tensor.py:297]: Don't change whitespace

File: test/test_tensor.py (hunk 1 above)
Comment (jla524) [reply test/test_tensor.py:67]: the test fails on GPU and I'm not really sure why.. below is the error message
```
self = <test.test_tensor.TestTinygrad testMethod=test_dropout>

    def test_dropout(self):
      Tensor.training = True
      n, rate = 1_000_000, 0.1
      w = Tensor.ones(n).dropout(rate)
      non_zeros = np.count_nonzero(w.data)
      expected = n * (1 - rate)
>     np.testing.assert_allclose(non_zeros, expected, rtol=1e-3)
E     AssertionError: 
E     Not equal to tolerance rtol=0.001, atol=0
E     
E     Mismatched elements: 1 / 1 (100%)
E     Max absolute difference: 899999.
E     Max relative difference: 0.99999889
E      x: array(1)
E      y: array(900000.)

test/test_tensor.py:73: AssertionError
```

**Comment by jla524** [comment]
I added `.cpu()` and the test case would pass for both CPU and GPU - sorry for being a noob earlier!

**Pull request merged**